AgContext.yaml

```diff
@@ -0,0 +1,149 @@
+
+agcontext_name:
+  type:
+    anyOf:
+    - type: string
+    description: |-
+      Agricultural context name
+      string. Required.
+      A string representing the name of the agricultural context.
+
+crop_type:
+  type:
+    anyOf:
+    - type: string
+    description: |-
+      General crop type
+      string. Required.
+      A string describing the general cropping scenario (e.g. "root vegetable").
+
+crop_stage:
+  type:
+    anyOf:
+    - type: string
+      enum:
+      - na
+      - 00
+      - 0
+      - 1
+      - 2
+      - 3
+      - 4
+      - 5
+      - 6
+      - 7
+      - 8
+      - 9
+      - 99
+    description: |-
+      BBCH crop development stage
+      "na", "00", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "99". Required.
+      One of eleven possible strings representing the BBCH phenological development stage of the crop. If this AgContext is not in a cropping environment, use value "na".
+
+location_continent:
+  type:
+    anyOf:
+    - type: string
+    description: |-
+      Continent where AgContext is located
+      string. Required.
+      A string representing the continent where the AgContext is located.
+
+location_country:
+  type:
+    anyOf:
+    - type: string
+    description: |-
+      Country where AgContext is located
+      string. Required.
+      A string representing the country where the AgContext is located.
+
+location_state:
+  type:
+    anyOf:
+    - type: string
+    description: |-
+      State where AgContext is located
+      string. Required.
+      A string representing the state/province where the AgContext is located.
+
+upload_time:
+  type:
+    anyOf:
+    - type: DateTime<'Y-m-d H:i:s'>
+    description: |-
+      Data and time when this AgContext was uploaded.
+      DateTime. Required.
+      System generated date time object recorded when a dataset with AgContext is uploaded.
+
+camera_type:
+  type:
+    anyOf:
+    - type: string
+    - type: string
+      enum:
+      - variable
+    description:
+      Type of camera
+      string.
+      A string describing the model/make of camera used for this AgContext. If this is not fixed within the AgContext, use value "variable".
```

Comment:
maybe we want this to be fixed within the agcontext, and force the user to create multiple agcontexts to describe multiple camera types??

Reply:
I really like that idea; it probably applies to all camera associated info.  camera_type for DeepWeeds is currently "weedlogger" which is Alex's name for his combination of lens, sensor, and computer. However for CWFID, I used the model name of the camera that they used. Should we also be including objects for camera_lens, etc.?